### ISSUE TITLE
Add inline mapping to SCVS BOM Maturity Model

### ISSUE THREAD

**Issue by stevespringett** [CycloneDX/specification#394]
A proposal has been suggested that the CycloneDX specification add native support for the [SCVS BOM Maturity Model](https://scvs.owasp.org/bom-maturity-model/) to the schema itself. This may likely be a JSON-only enhancement, but XML support may be considered.

For example, the definition of a `/components/[]/supplier` looks like:
```json
"supplier": {
  "title": "Component Supplier",
  "description": " The organization that supplied the component. The supplier may often be the manufacturer, but may also be a distributor or repackager.",
  "$ref": "#/definitions/organizationalEntity"
}
```

could be enhanced to:

```json
"supplier": {
  "title": "Component Supplier",
  "description": " The organization that supplied the component. The supplier may often be the manufacturer, but may also be a distributor or repackager.",
  "$ref": "#/definitions/organizationalEntity"
  "meta:scvs:version": "1.0",
  "meta:scvs:structure": "urn:owasp:scvs:bom:structure:inventory",
  "meta:scvs:taxonomy": [
    "urn:owasp:scvs:bom:provenance:role:supplier"
    "urn:owasp:scvs:bom:provenance:role:supplier:org",
    "urn:owasp:scvs:bom:provenance:role:supplier:non-org"
  ]
}
```

By supporting SCVS BOM Maturity Model in the schema itself, this would allow tool makers to leverage the model and have a build-in mapping. This will aid in:
- SBOM quality tools and score calculations
- Make policy and governance easier
  - [SCVS profiles](https://scvs.owasp.org/bom-maturity-model/profiles/) could be used together with CycloneDX natively
- Aid in conversions between BOM formats

**Comment by mrutkows** [comment]
@Mariuxdeangelo I hope to use/add a `"meta:scvs"` styled key to each data type in the schema (much like added for "meta:enum".

**Comment by jkowalleck** [comment]
I guess e could even grate an XSD extension that would allow adding these flags to out XSD files .... 
Not so sure about the protobuff schema files ...

**Comment by stevespringett** [comment]
Yeah, not too worried about protobuf. But if we find a way to do this in the XSD, that would be great. If not, the JSON Schema is our reference implementation for the specification and the Ecma standard, so its obviously a priority.

**Comment by jkowalleck** [comment]
> By supporting SCVS BOM Maturity Model in the schema itself, this would allow tool makers to leverage the model and have a build-in mapping. 

@stevespringett if we added those markers to a schema, how would this help anybody?
Is there any existing/discrete tool that would read those fields? If there is no such tool yet, where is the spec of such a tool - I mean i could probably come up with some alternatives to this very proposal, so that such a tool would still be functioning ... 

I find the proposal waaaay to vague, so I can not see any value in it yet. I would even say it is bloat I would avoid having in the schema files.

**Comment by stevespringett** [comment]
The CycloneDX SBOM Utility that Matt has created, currently uses Go annotations to do somthing similar. Its very manual however. The idea is to remove the annotations that are specific to the tool, and use the mapping provided by the schema itself.  In doing so, other tools would be able to adopt the same approach.

**Comment by jkowalleck** [comment]
>  In doing so, other tools would be able to adopt the same approach.

This would require other tools to follow the same annotation style we would come up now. 
I don't see that as a desirable way to work.


here is another approaches, which is more versatile in many means: 
Somebody provides mapping of each SCVS "flag" to (multiple) actual BOM-document targets in form of JSONPath(JSON and YAML documents), XPATH(XML documents), FieldMask(Protobuf documents). 

Such mapping is document centric (not schema- nor specification-centric).
Such mapping would not be part of any schema file, but independent.
Such mapping is portable.
Such mapping is machine readable.

example for CycloneDX JSON:
- `urn:owasp:scvs:bom:core:timestamp` -> `$.metadata.timestamp`
- `urn:owasp:scvs:bom:resource:software:sbom` -> `$..externalReferences[?($.type=='sbom')].url` 

This solution is actually one that the SCVS BOM Maturity Model WorkGroup should have come up with in the first place, since this is the most interoperable way (I can think of) of mapping their "Identifier" to an actual data structure. 
This is the relevant part: mapping to data structures, since they should not care for capabilities(schema or specification) but actual SBOM data.

These mappings can be provided by the SCVS community without any need for interaction with SBOM standards working groups, so this makes them even independent.

**Comment by stevespringett** [comment]
Closing this issue as it has been superseded by #736
